chain/chunks/src/lib.rs

```diff
@@ -1857,9 +1857,6 @@ impl ShardsManager {
             self.complete_chunk(&chunk_hash);
             return Ok(ProcessPartialEncodedChunkResult::HaveAllPartsAndReceipts);
         }
-
-        // add the chunk to the request pool in case it is not there already
-        self.request_chunk_single_mark_only(header);
```

Comment:
Thinking more carefully, I think we should call the `request_chunk_single_mark_only` function when processing partial encoded chunk messages (not forwards or response). In all other cases, the request has already been added to the pool and this line is redundant. For the partial encoded chunk case though, removing this line means the node will only start to request for chunks 400ms after the block is received, not 400ms after the chunk header is received.

Reply:
Ahhhhh I see. Great catch. Looks like I gotta add this back.